narwhal/primary/src/proposer.rs

```diff
@@ -83,11 +83,12 @@ pub struct Proposer {
     /// Holds the certificate of the last leader (if any).
     last_leader: Option<Certificate>,
     /// Holds the batches' digests waiting to be included in the next header.
-    digests: Vec<(BatchDigest, WorkerId, TimestampMs)>,
+    /// Digests are popped in LIFO order from the front.
+    digests: VecDeque<OurDigestMessage>,
```

Comment:
So whenever we retry we always want to retry the "latest/newest" batches first - meaning the ones from the higher rounds, correct?

Reply:
Sorry this is a leftover comment from a closed PR. Using LIFO was the previous idea, but I think it needs to be paired with updates to error handling and back pressure logic to work. So keeping the FIFO policy.